### PR TITLE
clean up book keeping object to avoid null pointers

### PR DESCRIPTION
**Pull request by Raynos**
Fixes: https://github.com/uber/hyperbahn/issues/61

r: @kriskowal @rf @jcorbin

### CODE REVIEW

File: service-proxy.js [hunk 1]
```diff
@@ -832,6 +832,8 @@ function reapPeers() {
         }
 
         self.channel.peers.delete(hostPort);
+        delete self.peersToReap[hostPort];
```
Comment (kriskowal) [inline service-proxy.js]: this should be unnecessary—peersToReap gets stomped on :839

File: service-proxy.js [hunk 2]
```diff
@@ -832,6 +832,8 @@ function reapPeers() {
...
+        delete self.connectedPeers[hostPort];
```
Comment (kriskowal) [inline service-proxy.js]: This should be unhelpful, because the peer cannot be reaped on the subsequent reaping. This introduces a leak.

Comment (Raynos) [inline service-proxy.js]: no ? Question mark. We just deleted it; i.e. `peers.delete(hp);`

**Comment by Raynos** [comment]
What this PR does is Once we remove a Peer object from memory we assume it is completely gone; i.e. `peers.delete(hp)` should remove the TChannelPeer object from memory completely.

After doing so I'm cleaning up the objects to remove a reference to this host port.

The only way it can come back is another open connection or `"ad"` i.e. the `refreshServicePeer()` method that get's called when we receive a `"relay-ad"`.

File: service-proxy.js (hunk 1 above)
Comment (Raynos) [inline service-proxy.js]: Yeah I think this is just me being silly.

**Comment by jcorbin** [comment]
lgtm now with less delet

**Comment by kriskowal** [comment]
:ok:

**Pull request merged**